src/api/extendOptions.ts

```diff
@@ -29,5 +30,9 @@ function getDefaultStorePath () {
 }
 
 export default (opts?: PnpmOptions): StrictPnpmOptions => {
+  opts = opts || {}
+  if (opts.flatTree === true && !preserveSymlinks) {
+    throw new TypeError('Flat tree is supported only on Node.js >= 6.3.0')
```

Comment:
Though, do we want to scream and stop work or warn and continue with non-flat tree?